### ISSUE TITLE
KCom2: Shipping providers and freight cost calculation

### ISSUE THREAD

**Issue by bradmac** [Kademi/kademi-dev#5489]
- support calculation based on weight or volumetric weight
- cols should be stored on ProductSku
- allow a list of shipping providers to be configured
- each provider can have a different calculation
- calculation can be standard or use a JS function

**Comment by bradmac** [comment]
Depends on https://github.com/Kademi/kademi-dev/issues/5497

**Comment by bradmac** [comment]
Or use this provider to support all carts
```
<map>
  <entry>
    <string>providers</string>
    <list>
      <default id="acme" title="ACME Freight and Logistics">
        <weightMult>1</weightMult>
        <volumetricMult>10</volumetricMult>
            <!--        
        <availableCountries class="java.util.Arrays$ArrayList">
          <a class="string-array">
            <string>NZ</string>
            <string>AU</string>
          </a>
        </availableCountries>
        <minTotalWeightKgs>10</minTotalWeightKgs>
        <maxTotalWeightKgs>100</maxTotalWeightKgs>
          -->                
      </default>
    </list>
  </entry>
</map>
```

**Comment by bradmac** [comment]
Blocked by https://github.com/Kademi/kademi-dev/issues/5497

**Comment by Uafinder** [comment]
It looks incorrectly count weight multiplier

http://vladtest54e.admin.kademi-ci.co/products/17947606/#skus-tab
product weight 5 kg
we select shipping provider with weight options
http://vladtest54e.admin.kademi-ci.co/ecommerce/vladtest54eweb-ecomstore/?shippingId=weight-amp-size-delivery
There is a multiplier on weight = 1.5
So, Freight cost should be 5x1.5 = 7.5$
But it shows incorrect value
![image](https://user-images.githubusercontent.com/1945230/47083797-6af19500-d21a-11e8-9d0e-dab1e55771f8.png)

**Comment by Uafinder** [comment]
Volumetric multiplier looks doesn't work at all
![image](https://user-images.githubusercontent.com/1945230/47084181-71344100-d21b-11e8-916d-1746ad8eed04.png)
![image](https://user-images.githubusercontent.com/1945230/47084193-7abda900-d21b-11e8-9bb5-0e33278a131f.png)
http://vladtest54e.admin.kademi-ci.co/products/17947604/#skus-tab
http://vladtest54e.admin.kademi-ci.co/ecommerce/vladtest54eweb-ecomstore/?shippingId=size-provider

**Comment by bradmac** [comment]
Closing, seems to be duplicate of https://github.com/Kademi/kademi-dev/issues/6174
